### PR TITLE
Add Proc.FileDescriptorTargets() function

### PR DESCRIPTION
**Pull request by nickygerritsen**
For a project we are creating we needed something that could scan /proc/ID/fd and tell us what target files are used for each process.

Before trying to code everything else I found this project which already did almost everything I wanted, so I added the missing function.

We are going to use it even if this would not get merged, but I think it is nice to at least propose a merge for something like this :). I'm of course willing to change the implementation if something should be done differently.

### CODE REVIEW

File: proc.go [hunk 1]
```diff
@@ -133,6 +133,28 @@ func (p Proc) FileDescriptors() ([]uintptr, error) {
...
+func (p Proc) FileDescriptorTargets() ([]string, error) {
+	names, err := p.fileDescriptors()
+	if err != nil {
+		return nil, err
+	}
+
+	var targets = make([]string, len(names))
+
+	for i, name := range names {
+		var target, err = p.readlink("fd/" + name)
```
Comment (juliusv) [inline proc.go]: Style nit: we do `:=` instead of `var` when the assignment happens on the same line: `target, err := ...`.

File: proc.go [hunk 2]
```diff
@@ -133,6 +133,28 @@ func (p Proc) FileDescriptors() ([]uintptr, error) {
...
+		if err != nil {
+			targets[i] = ""
```
Comment (juliusv) [inline proc.go]: It'll be the empty string by default, so this case could be omitted.

**Comment by juliusv** [comment]
:+1: Otherwise, provided you're ok with the API-breakage warning in the README.md.

**Comment by nickygerritsen** [comment]
OK I changed the code for both your comments and pushed again.

Furthermore we are probably going to create a fork that we will update when we have time to fix API-breakage stuff :)

**Comment by juliusv** [comment]
Thanks!

File: proc_test.go [hunk 3]
```diff
@@ -93,7 +93,54 @@ func TestFileDescriptors(t *testing.T) {
...
+	fdsBefore, err := p2.FileDescriptors()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s, err := os.Open("fixtures")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer s.Close()
```
Comment (grobie) [inline proc_test.go]: Why do you open fixtures, it doesn't seem to be used afterwards.

Comment (grobie) [inline proc_test.go]: Eh, ignore.

**Pull request merged**